### ISSUE TITLE
[BUG] Remote endpoint connector cannot parse array param

### ISSUE THREAD

**Issue by ps48** [opensearch-project/ml-commons#1036]
**What is the bug?**
Different Language models have different types of parameters that need to be passed in to the model endpoints. The connectors should support different data-types including array for such model.s

**How can one reproduce the bug?**
1. Use dev tools to create a model connector
2. API Sample with payload
```
POST /_plugins/_ml/connectors/_create
{
  "name": "Test connector",
  "description": "Test Connector",
  "version": 1,
  "protocol": "aws/v1",
  "access_mode": "public",
  "parameters": {
  "sample_param": [ "\n\nHuman:" ]
  },
  "credential": {
      "access_key": "<new access key>",
      "secret_key": "<new secret key>",
      "session_token": "<new session>"
  },
  "actions": [
    {
      "predict": {
        "method": "POST",
        "url": "URL",
        "headers": { 
          "content-type": "application/json",
          "x-amz-content-sha256": "required"
        },
        "request_body": "PARAMS"
      }
    },
    {
      "metadata": {
        "method": "GET",
        "url": "URL",
        "headers": {
          "content-type": "application/json"
        }
      }
    }
  ]
}

```
__NOTE__: The issue is with **sample_param** in the above payload

**What is the expected behavior?**
All parameters with different data-type should work out of the box with all APIs

**What is your host/environment?**
 - OS: MacOS
 - Version 2.8
 - Plugins 
    * ML Commons 2.8 from branch: [feature/remote-inference-rebase-2.x](https://github.com/opensearch-project/ml-commons/tree/feature/remote-inference-rebase-2.x)

**Do you have any screenshots?**

**Do you have any additional context?**
1. The issue may trickle down to model register and deploy APIs. Need to test more without connector and directly register/deploy models.

**Comment by Zhangxunmt** [comment]
This bug is fix in 2.9 in https://github.com/opensearch-project/ml-commons/pull/1075.

**Comment by ps48** [comment]
@Zhangxunmt @ylwu-amzn  Was this solved in 2.9? I still see the same error as earlier in the 2.9 version:
```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "Invalid JSON in payload"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "Invalid JSON in payload"
  },
  "status": 400
}
```
